bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/SingleProbeStrategy.scala

```diff
@@ -120,4 +125,34 @@ object SingleProbeStrategy {
     val angle  = atan2(sin(raDiff), cos(dec1) * tan(dec2) - sin(dec1) * cos(raDiff))
     Angle.fromRadians(angle)
   }
+
+  /**
+   * Given a list of candidates, bin them by quality and then pick the one that vignettes the
+   * science area the least.
+   * @param lst              the list of possible candidates
+   * @param mt               magnitude lookup table
+   * @param ctx              context information
+   * @param probe            the guide probe
+   * @param toSiderealTarget converts candidates to SiderealTargets
+   * @tparam A               the type of the candidates
+   * @return                 Some(guideStar) if a candidate exists that can be used, None otherwise
+   */
+  def brightestByQualityAndVignetting[A](lst: List[A], mt: MagnitudeTable, ctx: ObsContext, probe: ValidatableGuideProbe with VignettingGuideProbe)(toSiderealTarget: A => SiderealTarget): Option[A] = {
+    if (lst.isEmpty) None
```

Comment:
Do you need this case? If `lst` is empty then `candidates` will be empty and `reduceOption` will give you `None`.